test/services/storage/storage.spec.ts

```diff
@@ -0,0 +1,202 @@
+import chai from 'chai'
+import chaiAsPromised from 'chai-as-promised'
+import sinonChai from 'sinon-chai'
+import dirtyChai from 'dirty-chai'
+import sinon from 'sinon'
+import { EventData } from 'web3-eth-contract'
+import { asciiToHex } from 'web3-utils'
+import Eth from 'web3-eth'
+import { SubstituteOf } from '@fluffy-spoon/substitute'
+import { Sequelize } from 'sequelize-typescript'
+
+import eventProcessor from '../../../src/services/storage/storage.processor'
+import { OfferService, StorageServices } from '../../../src/services/storage'
+import { sequelizeFactory } from '../../../src/sequelize'
+import Offer from '../../../src/services/storage/models/offer.model'
+import { eventMock } from '../../utils'
+import { EventError } from '../../../src/errors'
+import BillingPlan from '../../../src/services/storage/models/price.model'
+
+chai.use(sinonChai)
+chai.use(chaiAsPromised)
+chai.use(dirtyChai)
+const expect = chai.expect
+
+describe('Storage services', function () {
+  const provider = 'TestAddress'
+  let sequelize: Sequelize
+  let eth: SubstituteOf<Eth>
+
+  before(() => {
+    sequelize = sequelizeFactory()
+  })
+
+  describe('Events Processor', () => {
+    describe('Offer events', () => {
+      let processor: (event: EventData) => Promise<void>
+      let offerService: OfferService
+      let offerServiceEmitSpy: sinon.SinonSpy
+
+      before(async () => {
+        offerService = new OfferService({ Model: Offer })
+        processor = eventProcessor({ offerService } as StorageServices, eth)
+        await sequelize.sync({ force: true })
```

Comment:
Hmm, this means that you are depending on the state from previous tests right? Eq. not starting from "scratch" every time?

I am strongly against this as this effectively requires running the whole test-suite every time which is something that is not guaranteed and actually even desired. If you have test-suite running for 10 minutes and you are trying debugging one test, you really don't want to run it all, but actually only that one test, which is something you can actually do using the `--grep` functionality of Mocha. E.g. `npm test -- -g "update BillingPlan"' which will fail even though the tested code might be correct.

Also, it is hard for understanding the tests because you have to read a lot of other tests to understand what state are you testing.

And also, if one test fails then it will break all dependant tests which might be actually pass otherwise.

So please rework the tests in the way that every test-case runs from scratch, setups the state (eq. DB etc.) before the running the tested code and then assert that the state changed as expected.

I know it is tedious and very verbose but it pays out to have it like this, at least for unit-tests, integration tests might be a different story.

Reply:
Good point, thanks 👍